nemo_rl/algorithms/sft.py

```diff
@@ -325,25 +364,23 @@ def validate(
                 )
             else:
                 if model_type == "lm":
-                    list_of_val_metrics.append(
-                        SFTValMetrics(val_loss=float(val_results["loss"]))
-                    )
+                    dict_val_metrics["val_loss"].append(float(val_results["loss"]))
                 elif model_type == "reward":
-                    list_of_val_metrics.append(
-                        RMValMetrics(
-                            val_loss=sum(val_results["all_mb_metrics"]["loss"]),
-                            accuracy=sum(val_results["all_mb_metrics"]["accuracy"]),
-                            rewards_chosen_mean=sum(
-                                val_results["all_mb_metrics"]["rewards_chosen_mean"]
-                            ),
-                            rewards_rejected_mean=sum(
-                                val_results["all_mb_metrics"]["rewards_rejected_mean"]
-                            ),
-                            num_valid_samples=sum(
-                                val_results["all_mb_metrics"]["num_valid_samples"]
-                            ),
-                        )
-                    )
+                    sum_num_valid_samples = sum(val_results["all_mb_metrics"]["num_valid_samples"])
+
+                    dict_val_metrics["val_loss"] += [
+                        value * sum_num_valid_samples for value in val_results["all_mb_metrics"]["loss"]
+                    ]
+                    dict_val_metrics["accuracy"] += [
+                        value * sum_num_valid_samples for value in val_results["all_mb_metrics"]["accuracy"]
+                    ]
+                    dict_val_metrics["rewards_chosen_mean"] += [
+                        value * sum_num_valid_samples for value in val_results["all_mb_metrics"]["rewards_chosen_mean"]
+                    ]
+                    dict_val_metrics["rewards_rejected_mean"] += [
+                        value * sum_num_valid_samples for value in val_results["all_mb_metrics"]["rewards_rejected_mean"]
+                    ]
```

Comment:
Can this be simplified with something like
```suggestion
                    for k in ["loss", "accuracy", "rewards_chosen_mean", "rewards_rejected_mean"]:
                        dict_val_metrics[k if k != "loss" else "val_loss"] += [
                            value * sum_num_valid_samples for value in val_results["all_mb_metrics"][k]
                        ]
```

Reply:
Yes, I updated